tests/unit/bundle_analysis/test_asset_association.py

```diff
@@ -0,0 +1,92 @@
+from pathlib import Path
+from typing import Dict
+from unittest.mock import patch
+
+import pytest
+
+from shared.bundle_analysis import BundleAnalysisReport, BundleAnalysisReportLoader
+from shared.bundle_analysis.models import Asset, AssetType, MetadataKey
+from shared.storage.exceptions import PutRequestRateLimitError
+from shared.storage.memory import MemoryStorageService
+
+bundle_stats_prev_a_path = (
+    Path(__file__).parent.parent.parent / "samples" / "asset_link_prev_a.json"
+)
+
+bundle_stats_prev_b_path = (
+    Path(__file__).parent.parent.parent / "samples" / "asset_link_prev_b.json"
+)
+
+bundle_stats_curr_a_path = (
+    Path(__file__).parent.parent.parent / "samples" / "asset_link_curr_a.json"
+)
+
+bundle_stats_curr_b_path = (
+    Path(__file__).parent.parent.parent / "samples" / "asset_link_curr_b.json"
+)
+
+
+def _get_asset_mapping(
+    bundle_analysis_report: BundleAnalysisReport, bundle_name: Asset
+) -> Dict[str, str]:
+    bundle_report = bundle_analysis_report.bundle_report(bundle_name)
+    asset_report = list(bundle_report.asset_reports())
+    return {asset.hashed_name: asset for asset in asset_report}
+
+
+def test_asset_association():
+    try:
```

Comment:
Why have a `try:` without an `except`?

"to have the `finally;` block" 
but what's the difference of having it at the end of the try block?